### ISSUE TITLE
Add debug configuration to build matrix on Appveyor

### ISSUE THREAD

**Issue by michaeltryby** [OpenWaterAnalytics/EPANET#435]
We could certainly use a Debug configuration build on the CI worker to help flag memory issues before they get merged.

**Comment by LRossman** [comment]
Why is the debug build for Win64 and not Win32 like the release build?

**Comment by michaeltryby** [comment]
It seems like  some memory bugs have crept in to the code base. The thought was to add a debug build to the CI to make them easier to identify. I chose the 64 bit build because it is experimental (i.e. failures are allowed). Yes I am using  crtdbg library. So what exactly is your concern?

**Comment by michaeltryby** [comment]
Also. The PR isn’t finished yet.

**Comment by LRossman** [comment]
FYI - when I was building some of the unit tests in stand-alone fashion (i.e., without using Boost) under the CRT library I got some small memory leaks reported when I combined the unit test .cpp file with the toolkit's .c files. If I gave the test file a .c extension and re-built the leaks no longer appeared. Not sure why this happened. Could it be related to how CRT treats the C++ `new `operator as discussed [here](https://docs.microsoft.com/en-us/visualstudio/debugger/finding-memory-leaks-using-the-crt-library?view=vs-2017)?
